src/azure-cli-core/azure/cli/core/auth/identity.py

```diff
@@ -318,20 +318,13 @@ def build_credential(cls, secret_or_certificate=None,
         }
         """
         entry = {}
-        if certificate:
+        if client_secret:
+            logger.warning(PASSWORD_CERTIFICATE_WARNING)
```

Comment:
Because the `os.path.isfile()` detection logic is removed, it is no longer possible to know if `client_secret` is a certificate, so the warning is now unconditionally shown when `--password` is used.